lib/Dialect/LLHD/Simulator/Engine.cpp

```diff
@@ -80,10 +81,10 @@ int Engine::simulate(int n) {
   int i = 0;
 
   // Keep track of the instances that need to wakeup.
-  std::vector<std::string> wakeupQueue;
+  llvm::SmallSet<std::string, 8> wakeupQueue;
```

Comment:
At some point we'll probably want to use some string interning / ID-based identifiers to make this faster at runtime. But I guess if you haven't seen this crop up in the perf/flamegraph of the simulator, it's not really a pressing issue 😃

Reply:
I already have an integer-indexed version together (more similar to the signals indexing) with the other optimizations (it was indeed taking quite some chunk of the runtime), I'll upstream those in a separate PR :)

Reply:
Cool stuff!